tiledb/sm/filesystem/posix.cc

```diff
@@ -347,6 +347,29 @@ Status Posix::copy_file(
   return Status::Ok();
 }
 
+Status Posix::copy_dir(
+    const std::string& old_path, const std::string& new_path) {
+  RETURN_NOT_OK(create_dir(new_path));
+  std::vector<std::string> paths;
+  ls(old_path, &paths);
+  while (!paths.empty()) {
+    std::string file_name_abs = paths.front();
+    std::string file_name = file_name_abs.substr(old_path.length() + 1);
+    paths.erase(paths.begin());
+    if (is_dir(file_name_abs)) {
+      RETURN_NOT_OK(create_dir(new_path + "/" + file_name));
+      std::vector<std::string> child_paths;
+      ls(file_name_abs, &child_paths);
```

Comment:
Any call returning a `Status` object needs to be checked, please add a `RETURN_NOT_OK` here as well and `copy_file`.